tools/cli/domainCommands.go

```diff
@@ -681,6 +773,398 @@ func (d *domainCLIImpl) describeDomain(
 	return d.domainHandler.DescribeDomain(ctx, request)
 }
 
+func (d *domainCLIImpl) migrateDomain(c *cli.Context) {
+	var results []DomainMigrationRow
+	checkers := []func(*cli.Context) DomainMigrationRow{
+		d.migrationDomainMetaDataCheck,
+		d.migrationDomainWorkFlowCheck,
+		d.migrationDynamicConfigCheck,
+		d.searchAttributesChecker,
+	}
+	wg := &sync.WaitGroup{}
+	for i := range checkers {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			result := checkers[i](c)
+			results = append(results, result)
+		}(i)
+	}
+	wg.Wait()
+
+	renderOpts := RenderOptions{
+		DefaultTemplate: newtemplateDomain,
+		Color:           true,
+		Border:          true,
+		PrintDateTime:   true,
+	}
+
+	if err := Render(c, results, renderOpts); err != nil {
+		ErrorAndExit("Failed to render", err)
+	}
+}
+
+func (d *domainCLIImpl) migrationDomainMetaDataCheck(c *cli.Context) DomainMigrationRow {
+	d.destinationClient = newClientFactory(func(c *cli.Context) string {
+		return c.String(FlagDestinationAddress)
+	}).ServerFrontendClient(c)
+	domain := c.GlobalString(FlagDomain)
+	newDomain := c.String(FlagDestinationDomain)
+	request := &types.DescribeDomainRequest{
+		Name: &domain,
+	}
+	request2 := &types.DescribeDomainRequest{
+		Name: &newDomain,
+	}
+	ctx, cancel := newContext(c)
+	defer cancel()
+
+	currResp, err := d.frontendClient.DescribeDomain(ctx, request)
+
+	if err != nil {
+		ErrorAndExit(fmt.Sprintf("Could not describe old domain, Please check to see if old domain exists before migrating."), err)
+	}
+	newResp, err := d.destinationClient.DescribeDomain(ctx, request2)
+	if err != nil {
+		ErrorAndExit(fmt.Sprintf("Could not describe new domain, Please check to see if new domain exists before migrating."), err)
+	}
+	validationResult := metaDataValidation(currResp, newResp)
+	validationRow := DomainMigrationRow{
+		ValidationCheck:  "Domain Meta Data",
+		ValidationResult: validationResult,
+		ValidationDetails: ValidationDetails{
+			CurrentDomainRow: currResp,
+			NewDomainRow:     newResp,
+		},
+	}
+	return validationRow
+}
+
+func metaDataValidation(currResp *types.DescribeDomainResponse, newResp *types.DescribeDomainResponse) bool {
+	if !reflect.DeepEqual(currResp.Configuration, newResp.Configuration) {
+		return false
+	}
+
+	if currResp.DomainInfo.OwnerEmail != newResp.DomainInfo.OwnerEmail {
+		return false
+	}
+	return true
+}
+
+func (d *domainCLIImpl) migrationDomainWorkFlowCheck(c *cli.Context) DomainMigrationRow {
+	d.destinationClient = newClientFactory(func(c *cli.Context) string {
+		return c.String(FlagDestinationAddress)
+	}).ServerFrontendClient(c)
+	countWorkFlows := d.countLongRunningWorkflowinDest(c)
+	if countWorkFlows > 0 {
+		ErrorAndExit(fmt.Sprintf("Number of workflows is > 0"), nil)
+	}
+	return DomainMigrationRow{
+		ValidationCheck:  "Workflow Check",
+		ValidationResult: true,
+		ValidationDetails: ValidationDetails{
+			LongRunningWorkFlowNum: &countWorkFlows,
+		},
+	}
```

Comment:
Instead of ErrorAndExit, maybe return false check
```suggestion
        check := countWorkFlows == 0
	return DomainMigrationRow{
		ValidationCheck:  "Workflow Check",
		ValidationResult: check,
		ValidationDetails: ValidationDetails{
			LongRunningWorkFlowNum: &countWorkFlows,
		},
	}
```